src/arch/xtensa/CMakeLists.txt

```diff
@@ -77,7 +77,7 @@ target_compile_options(sof_options INTERFACE ${stdlib_flag} -fno-inline-function
 #      better to have set of default flags and change it only for special cases
 #   3) custom function that is used instead of target_sources and sets flags
 #      for each added source based on file extension
-target_compile_options(sof_options INTERFACE $<$<COMPILE_LANGUAGE:C>:-O2 -g -Wall -Werror -Wl,-EL -Wmissing-prototypes -Wpointer-arith -mtext-section-literals>)
+target_compile_options(sof_options INTERFACE $<$<COMPILE_LANGUAGE:C>:-Os -g -Wall -Werror -Wl,-EL -Wmissing-prototypes -Wpointer-arith -mtext-section-literals>)
```

Comment:
I don't know if it's good idea to optimize for size in every case - it will impact performance globally.
I thought you just gonna change that flag for BYT & XCC combo.

@lgirdwood @tlauda @mmaka1 @lbetlej @dbaluta 
What is our priority - speed or size?

Reply:
We can also move decision to Kconfig and have CONFIG_OPTIMIZE_SIZE that changes this flag to -Os and add it to baytrail_defconfig, I think it's actually the best way so far to handle this case.

Reply:
@xiulipan let me create PR which introduces this option

Reply:
@jajanusz sure, I will try to rebase the code with your change.

Reply:
@jajanusz For platforms with very little memory such as Baytrail size is probably more important (you do need to fit within the constraints). For i.MX the priority would be speed (we have 16MB of DRAM which is actually mostly wasted for right now, no need to optimize the code for size but we want the best speed possible). I believe the decision ought to be done on a case-by-case basis for each of the platforms.

Reply:
@xiulipan priority is always speed.